src/wasm/wasm-binary.cpp

```diff
@@ -5599,21 +5599,27 @@ bool WasmBinaryBuilder::maybeVisitRefTest(Expression*& out, uint32_t code) {
   if (code != BinaryConsts::RefTest) {
     return false;
   }
-  auto* curr = allocator.alloc<RefTest>();
-  WASM_UNREACHABLE("TODO (gc): ref.test");
-  curr->finalize();
-  out = curr;
+  auto heapType1 = getHeapType();
+  auto heapType2 = getHeapType();
```

Comment:
So instead of taking one static heap type this instruction takes two static heap types and a dynamic heap type!? I must be missing something big here...

Reply:
It does seem excessive, yeah. Not sure what you mean by 2 and 1, though? It gets two arguments that have a type with a heap type, and it also gets those heap types as immediates.

Reply:
The one "dynamic heap type" I was thinking about was the rtt value. We certainly couldn't get rid of the reference typed argument!

Reply:
I think we also can't get rid of the rtt - the VM checks that value at runtime. So both the ref and rtt seem equally necessary?